BitwardenShared/UI/Platform/Settings/Settings/SettingsProcessor.swift

```diff
@@ -1,32 +1,80 @@
+// MARK: - SettingsProcessorDelegate
+
+/// A delegate of `SettingsProcessor` that is notified when the settings tab badge needs to be updated.
+///
+@MainActor
+protocol SettingsProcessorDelegate: AnyObject {
+    /// Called when the settings tab badge needs to be updated.
+    ///
+    /// - Parameter badgeValue: The value to display in the settings tab badge.
+    ///
+    func updateSettingsTabBadge(_ badgeValue: String?)
+}
+
 // MARK: - SettingsProcessor
 
 /// The processor used to manage state and handle actions for the settings screen.
 ///
 final class SettingsProcessor: StateProcessor<SettingsState, SettingsAction, Void> {
     // MARK: Types
 
-    typealias Services = HasErrorReporter
-        & HasSettingsRepository
+    typealias Services = HasConfigService
+        & HasErrorReporter
+        & HasStateService
 
     // MARK: Private Properties
 
+    /// The task used to update the tab's badge count.
+    private var badgeUpdateTask: Task<Void, Never>?
+
     /// The `Coordinator` that handles navigation.
     private let coordinator: AnyCoordinator<SettingsRoute, SettingsEvent>
 
+    /// A delegate of the processor that is notified when the settings tab badge needs to be updated.
+    private weak var delegate: SettingsProcessorDelegate?
+
+    /// The services used by this processor.
+    private var services: Services
+
     // MARK: Initialization
 
     /// Creates a new `SettingsProcessor`.
     ///
     /// - Parameters:
     ///   - coordinator: The `Coordinator` that handles navigation.
+    ///   - delegate: A delegate of the processor that is notified when the settings tab badge needs
+    ///     to be updated.
+    ///   - services: The services used by the processor.
     ///   - state: The initial state of the processor.
     ///
     init(
         coordinator: AnyCoordinator<SettingsRoute, SettingsEvent>,
+        delegate: SettingsProcessorDelegate,
+        services: Services,
         state: SettingsState
     ) {
         self.coordinator = coordinator
+        self.delegate = delegate
+        self.services = services
         super.init(state: state)
+
+        // Kick off this task in init so that the tab bar badge will be updated immediately when
+        // the tab bar is shown vs once the user navigates to the settings tab.
+        badgeUpdateTask = Task { @MainActor [weak self] in
```

Comment:
just for my understanding, whats the benefit of storing this task in a variable vs just kicking off the Task in the init?

Reply:
Storing a reference to the task allows us to cancel it when the processor goes away (e.g. when you switch accounts or logout, etc).

Reply:
makes sense, thanks!